torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -173,6 +222,11 @@ void DistEngine::computeDependencies(
     for (const auto& recvBackwardEdge : recvBackwardEdges) {
       graphTask->exec_info_[recvBackwardEdge.function.get()].needed_ = true;
     }
+    // Mark all output nodes as needed.
+    for (const auto& outputEdge : outputEdges) {
+      const auto func = outputEdge.function.get();
+      graphTask->exec_info_[func].needed_ = true;
```

Comment:
All the exec_info_ can be removed now. Because you want the whole graph to be processed.